src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -626,9 +626,7 @@ int JVM_HANDLE_XXX_SIGNAL(int sig, siginfo_t* info,
 
   // Check for UD trap caused by NOP patching.
   // If it is, patch return address to be deopt handler.
-  if (!signal_was_handled) {
-    address pc = os::Posix::ucontext_get_pc(uc);
-    assert(pc != NULL, "");
+  if (!signal_was_handled && pc != NULL && os::is_readable_pointer(pc)) {
```

Comment:
If these new conditions are not met then what happens? What code will get the unreadable, or null pc to process?

Reply:
It will fall through and treat the signal as unhandled, resulting in a crash dump.  The old code was wrong for these reasons:

- It would assert in the signal handler if PC was 0
- It would crash in the signal handler if PC was not in readable memory
- It ignored the previously adjusted value of PC, which means trap-based NOP patching probably wouldn't have worked on platforms that need the PC adjusted, like s390